pkg/operator/k8sutil/node.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2018 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package k8sutil for Kubernetes helpers.
+package k8sutil
+
+import (
+	"fmt"
+
+	rookalpha "github.com/rook/rook/pkg/apis/rook.io/v1alpha2"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	helper "k8s.io/kubernetes/pkg/api/v1/helper"
+)
+
+func ValidNode(node v1.Node, placement rookalpha.Placement) (bool, error) {
```

Comment:
Could you add a unit tests around detecting invalid nodes? Perhaps if you modified the `createNode` helper in the unit test above, it could also create a node that is NotReady or has a taint.

Reply:
added